### PR TITLE
make docker-checker more robust

### PR DESCRIPTION
**Pull request by ArtfulCoder**
Partly addresses: #23556 by forcefully kill problematic docker processes.

### CODE REVIEW

**Comment by ArtfulCoder** [comment]
partly addresses  #23556

File: cluster/saltbase/salt/supervisor/docker-checker.sh [hunk 1]
```diff
@@ -35,7 +46,7 @@ echo "waiting 30s for startup"
 sleep 30
 
 while true; do
-  if ! sudo timeout 10 docker version > /dev/null; then
+  if ! sudo timeout 10 docker ps > /dev/null; then
```
Comment (yujuhong) [inline cluster/saltbase/salt/supervisor/docker-checker.sh]: 10 second is not enough for `docker ps` at heavy load based on the past experiences (we changed it to `docker ps` and then back to `docker version` once). We'd need a longer timeout. cc @dchen1107

Comment (ArtfulCoder) [inline cluster/saltbase/salt/supervisor/docker-checker.sh]: Also, I am not sure why we are using sudo.

Comment (dchen1107) [inline cluster/saltbase/salt/supervisor/docker-checker.sh]: yes, please increase the timeout value to 30 second here. sudo can be removed.

File: cluster/saltbase/salt/supervisor/docker-checker.sh [hunk 2]
```diff
@@ -20,11 +20,22 @@
...
 # Make sure docker gracefully terminated before start again
+starttime=`date +%s`
 while pidof docker > /dev/null; do
-    echo "waiting clean shutdown"
-    sleep 10
+    currenttime=`date +%s`
+    ((elapsedtime = currenttime - starttime))
+    # after 60 seconds, forcefully terminate docker process
+    if test $elapsedtime -gt 60; then
+      echo "attempting to kill docker process with sigterm signal"
```
Comment (dchen1107) [inline cluster/saltbase/salt/supervisor/docker-checker.sh]: s/sigterm/sigkill?

**Comment by ArtfulCoder** [comment]
PTAL

**Comment by dchen1107** [comment]
LGTM

**Comment by a-robinson** [comment]
Manually merging to reduce flakiness

**Comment by david-mcmahon** [comment]
The cherrypick process is defined here.  Hoping to get this PR merged today, but the details are [here](https://github.com/kubernetes/kubernetes/pull/23555/files).  You need to run the cherry pick tool yourself to get these onto the branch.  @bgrant0607 will then approve them for merge.

**Comment by bgrant0607** [comment]
Every cherrypick needs a release note.

**Comment by bgrant0607** [comment]
This PR has no description and no reference to a P0 issue.

**Comment by cjcullen** [comment]
Fixes the "docker flakes on nodes and masters" part of #23556.

**Comment by cjcullen** [comment]
Internal bugref: 27875273

**Comment by dchen1107** [comment]
@bgrant0607 #23556 is p0 issue. I have a small summary related to the issue at: https://github.com/kubernetes/kubernetes/issues/23556#issuecomment-203697231

**Pull request merged**